### PR TITLE
Docs: PIC Model

### PR DESCRIPTION
**Pull request by ax3l**
Start writing the PIC model section. Introduces our implemented system of equations for the reader.

ccing @ameresj 

Depending on how you like this, we can merge a first draft and you guys can contribute further updates via additional PRs or we improve it already in this PR.

Note: docs on how to build these docs locally: http://picongpu.readthedocs.io/en/latest/dev/sphinx.html

### CODE REVIEW

File: docs/source/models/pic.rst [hunk 1]
```diff
@@ -7,9 +7,81 @@ The Particle-in-Cell Algorithm
...
+These markers carry a spatial shape of order :math:`n` and a delta-distribution in momentum space.
+In most cases, these shapes are implemented as B-splines and are pre-integrated to *assignment functions* :math:`S` of the form:
+
+.. math::
+
+   S^0(x) = \big\{ \substack{1 \qquad \text{if}~0 \le x \lt 1\\ 0 \qquad \text{else}}
+
+   S^n(x) = \left(S^{n-1} * S^0\right)(x) = \int_{x-1}^x S^{n-1}(\xi) d\xi
+
+PIConGPU implements these up to order :math:`n=4`.
```
Comment (steindev) [inline docs/source/models/pic.rst]: Plus `The three dimensional marker shape is a multiplicative union of B-splines S(x,y,z) = S(x)S(y)S(z)`?

**Comment by ax3l** [comment]
@steindev thx for the review, I added your suggestions :)

**Comment by ameresj** [comment]
multiplicative union = tensor product, so tensor product B-Splines

I will have to include: the relativistic vlasov equation with characteristics, overview over normalization (eventually from my thesis) and an introduction into the sampling.

**Comment by ax3l** [comment]
@ameresj if you like we can merge this `PR` already to `dev` and you can update it when you have the time for it with a new PR.

**Pull request merged**